docs/weave-tutorial/07-handlers.md

```diff
@@ -0,0 +1,675 @@
+				"Content":      nil,
+				"CommentCount": nil,
+				"LikeCount":    nil,
+				"CreatedAt":    nil,
+				"DeleteAt":     nil,
+			},
+			wantDeliverErrs: map[string]*errors.Error{
+				"Metadata":     nil,
+				"ID":           nil,
+				"BlogID":       nil,
+				"Owner":        nil,
+				"Title":        nil,
+				"Content":      nil,
+				"CommentCount": nil,
+				"LikeCount":    nil,
+				"CreatedAt":    nil,
+				"DeleteAt":     nil,
+			},
+		},
+		"failure missing signer": {
+			msg: &CreateArticleMsg{
+				Metadata: &weave.Metadata{Schema: 1},
+				BlogID:   ownedBlog.ID,
+				Title:    "insanely good title",
+				Content:  "best content in the existence",
+				DeleteAt: future,
+			},
+			expected: nil,
+			wantCheckErrs: map[string]*errors.Error{
+				"Metadata":     nil,
+				"ID":           nil,
+				"BlogID":       nil,
+				"Owner":        nil,
+				"Title":        nil,
+				"Content":      nil,
+				"CommentCount": nil,
+				"LikeCount":    nil,
+				"CreatedAt":    nil,
+				"DeleteAt":     nil,
+			},
+			wantDeliverErrs: map[string]*errors.Error{
+				"Metadata":     nil,
+				"ID":           nil,
+				"BlogID":       nil,
+				"Owner":        nil,
+				"Title":        nil,
+				"Content":      nil,
+				"CommentCount": nil,
+				"LikeCount":    nil,
+				"CreatedAt":    nil,
+				"DeleteAt":     nil,
+			},
+		},
+	}
+	for testName, tc := range cases {
+		t.Run(testName, func(t *testing.T) {
+			auth := &weavetest.Auth{
+				Signer: tc.signer,
+			}
+
+			// initalize environment
+			rt := app.NewRouter()
+			RegisterRoutes(rt, auth)
+			kv := store.MemStore()
+
+			// initalize blog bucket and save blogs
+			blogBucket := NewBlogBucket()
+			err := blogBucket.Put(kv, ownedBlog)
+			assert.Nil(t, err)
+
+			err = blogBucket.Put(kv, notOwnedBlog)
+			assert.Nil(t, err)
+
+			// initialize article bucket
+			articleBucket := NewArticleBucket()
+
+			tx := &weavetest.Tx{Msg: tc.msg}
+
+			ctx := weave.WithBlockTime(context.Background(), time.Now().Round(time.Second))
+
+			if _, err := rt.Check(ctx, kv, tx); err != nil {
+				for field, wantErr := range tc.wantCheckErrs {
+					assert.FieldError(t, err, field, wantErr)
+				}
+			}
+
+			res, err := rt.Deliver(ctx, kv, tx)
+			if err != nil {
+				for field, wantErr := range tc.wantDeliverErrs {
+					assert.FieldError(t, err, field, wantErr)
+				}
+			}
+
+			if tc.expected != nil {
+				var stored Article
+				err := articleBucket.One(kv, res.Data, &stored)
+				assert.Nil(t, err)
+
+				// ensure createdAt is after test execution starting time
+				createdAt := stored.CreatedAt
+				weave.InTheFuture(ctx, createdAt.Time())
+
+				// avoid registered at missing error
+				tc.expected.CreatedAt = createdAt
+
+				assert.Equal(t, tc.expected, &stored)
+			}
+		})
+	}
+}
+```
+
+Test case above has quite amount of necessary boiler plate. Every test case including cases that tests missing message fields which depends on message validation. Usual test work flow is:
```

Comment:
`Test case above has quite an amount of necessary boiler plate. Every test case including cases that test missing message fields which depends on message validation. The usual test work flow is:
`